language/stdlib/modules/approved_payment.move

```diff
@@ -0,0 +1,95 @@
+// Module that allows a payee to approve payments with a cryptographic signature. The basic flow is:
+// (1) Payer sends `metadata` to the payee
+// (2) Payee signs `metadata` and sends 64 byte signature back to the payer
+// (3) Payer sends an approved payment to the payee by sending a transaction invoking `deposit`
+//     with payment data + signature. The transaction will abort if the signature check fails.
+// Note: approved payment's are an accounting convenience/a courtesy mechansim for the payee, *not*
+// a hurdle that must be cleared for all payments to the payee. In addition, approved payments do
+// not have replay protection.
+address 0x0:
+module ApprovedPayment {
+    use 0x0::Libra;
+    use 0x0::LibraAccount;
+    use 0x0::Signature;
+    use 0x0::Transaction;
+    use 0x0::Vector;
+
+    // A resource to be published under the payee's account
+    resource struct T {
+        // 32 byte EdDSA public key whose counterpart must be used to sign the payment data
```

Comment:
Let’s mention here “single” Ed25519 key. 
I guess we don’t want to give optionality as in auth_key, right?
Also, maybe mention “unlike auth_key which uses hashed public keys”

Reply:
Good question on optionality. It seems simpler to stick with just Ed25519, but would definitely be easy to add flexibility if there's a need for multiple schemes.

Reply:
Optionality is not required for now. I guess the main question will always be "why raw PK here Vs hashing at the auth_key layer".